### PR TITLE
*: add admin permission checking

### PR DESCRIPTION
**Pull request by xiang90**
/cc @mitake

### CODE REVIEW

File: auth/store.go [hunk 1]
```diff
@@ -699,3 +712,12 @@ func NewAuthStore(be backend.Backend) *authStore {
 		be: be,
 	}
 }
+
+func hasRootRool(u *authpb.User) bool {
```
Comment (mitake) [inline auth/store.go]: hasRootRole?

File: etcdserver/apply.go [hunk 2]
```diff
@@ -710,3 +718,38 @@ func compareInt64(a, b int64) int {
...
 	return len(rangeEnd) == 1 && rangeEnd[0] == 0
 }
+
+func needAdminPermission(r *pb.InternalRaftRequest) bool {
+	switch {
+	case r.AuthEnable != nil:
+		return true
+	case r.AuthDisable != nil:
+		return true
+	case r.Authenticate != nil:
```
Comment (mitake) [inline etcdserver/apply.go]: Authenticate must be allowed to every user

File: auth/store.go (hunk 1 above)
Comment (xiang90) [inline auth/store.go]: good catch. will fix.

**Comment by mitake** [comment]
@xiang90 how about making other operations, e.g. membership change, admin ones? They shouldn't be done by ordinal users.

File: etcdserver/apply.go (hunk 2 above)
Comment (xiang90) [inline etcdserver/apply.go]: oh... sorry.

**Comment by xiang90** [comment]
@mitake For membership, yes.

**Comment by xiang90** [comment]
@mitake The membership commands not do not go through v3 apply loop. I will make it happen in another pr.

**Comment by mitake** [comment]
@xiang90 sure, supporting the membership commands can be done later

**Comment by xiang90** [comment]
@mitake I fixed the typo. PTAL.

**Comment by mitake** [comment]
@xiang90 thanks, lgtm after CI pass

**Pull request merged**